.travis.yml

```diff
@@ -42,58 +42,59 @@ jobs:
 # ---------------------------------------
     - stage: Unit tests & Linters
       name: "SonarQube"
+      if: fork != true
 
-      if: type = pull_request && fork != true
-
+      os: linux
+      sudo: required
       language: node_js
+
       node_js:
         - "8"
 
-      addons:
-        sonarqube: true
-
-      install:
-        - npm install --silent --only=dev
-
-      script:
-        - bash .ci/sonar.sh
-
-# ---------------------------------------
-# ESLint
-# ---------------------------------------
-    - stage: Unit tests & Linters
-      name: "ESLint"
+      cache:
+        directories: 
+          - $HOME/.sonar/cache
 
-      language: node_js
-      node_js:
-        - "8"
+      env:
+        # SonarQube Token
+        secure: KWhqHGlzE1ioGyZzcWxoTdhJdSnXvORHIZXvHC7WRdnOLa14gTdRkdjezzVT2DIeHcqH5x5uffAaWHgXWYPzqgm64XpR86UAcQeqf5FxF7V91M0jIaBlktzTK7Xf8PGozh0hqJqD4lTcG4hLW99a/dI0Z/LULape5dNBZWHKGwk6XjYY7rQErtSw1l4jrZD2w6xEVjXQSckwFVnbF5whoYiwOJwYzyNnfrY6NNWHtTwzSKmdK1DUQr6Vk+eA+ggndWXe2GA3bJGM9K4lNbG/TcTLUTULOfkqwoZVejOrd5dpuNZqa/iRPubtVuFUZ0W0+DFGl8KBctdkbhqI4qk27uQPIVQO7zF36hR1oaQuSNXyXwWOVtVm3+ymu1YAfzjVUTiRBHg5Uda0Ink8yLo8Wpnp0gRq5Y3iVhymMYbiKHuzs52tnHfZ6NFjIH1M86uZhzrJhCqFKKCX9dzguj1PrM+7zYC0aXIxi3Yt6k1C2DY/8A+g201tBhMB7EwppilH6Rm2sqBXpUKI/P/LW/lJOUmY4EGBOPv+RGX0W3Av28NTI0PisCiNKgD8w5r+4l16/AhMBY3lmvTjZFd6IpfcJohWA25k2h7b4Opemu7/po7faMGMhqUTW64D98YhqiWtR0QpVtIRnLiol4GbwwUjT83p1Nw+korROXsei5ygX/w=
 
       addons:
         sonarqube: true
 
       install:
-        - npm install --silent
         - npm install --silent --only=dev
 
       script:
-        - npm run --silent lint
+        - bash .ci/sonar.sh
 
 # ---------------------------------------
 # Functionnal Testing
 # ---------------------------------------
 # Node 6 - x64
 # ---------------------------------------
-    - stage: Functional tests
-      name: "Functionnal Tests - amd64 - Node v6"
+    - stage: "E2E Tests"
+      name: "amd64 - Node v6"
+
+      if: |
+        (type = push AND branch =~ ^(master|[0-9]+-dev)$) OR
+        (type = pull_request AND branch = master) OR
```

Comment:
Question: Shouldn't we still run the e2e tests with node v6 even on pull request?